openedx_events/learning/data.py

```diff
@@ -493,3 +520,74 @@ class ORASubmissionData:
     created_at = attr.ib(type=datetime)
     submitted_at = attr.ib(type=datetime)
     answer = attr.ib(type=ORASubmissionAnswer)
+
+
+@attr.s(frozen=True)
+class CoursePassingStatusData:
+    """
+    Represents the event data when a user's grade is updated, indicates if current grade is enough for course passing.
+
+    Attributes:
+        status (str): A string containing information about user's current course grade value
+        in comparison to the grading policy threshold.
+        user (UserData): An instance of UserData containing information about the user whose grade was updated.
+        course (CourseData): An instance of CourseData containing details about the course
+        in which the grade was updated.
+    """
+
+    status = attr.ib(type=str)
```

Comment:
Picking up the conversation about whether using `status='passing' | 'failing'` or `is_passing=True':

#### Option 1 `status='passing' | 'failing'`:

**Pros**
- Explicit status attribute for `CoursePassingStatus...`

**Cons**
- Hard-coded constant
- No validation support yet. Status could be anything.
- The integration PR would need to know the status format:
```
        CCX_COURSE_PASSING_STATUS_UPDATED.send_event(
            course_passing_status=CcxCoursePassingStatusData(
                status='passing',
                user=UserData(
                    pii=UserPersonalData(
                        username=user.username,
                        email=user.email,
                        name=user.get_full_name(),
                    ),
                    id=user.id,
                    is_active=user.is_active,
                ),
                course=CcxCourseData(
                    ccx_course_key=course_id,
                    master_course_key=course_id.to_course_locator(),
                ),
            )
        )
```

`status='passing'` was previously `CcxCoursePassingStatusData.PASSING`, which was standardized but then dropped altogether with the validator, but for maintainability reasons, we shouldn't use `status='passing'`.


#### Option 2 `is_passing=True`:

**Pros**
- It's similar to what's currently used by the PersistentCourseGrade model to send the events: https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/grades/course_grade_factory.py#L180 `passed=grade.passed`
- Doesn't need validation, it could be either True or False


**Cons**
- Is it explicit enough?

Which of the options should we choose? I'm leaning towards option 2, but you folks have more context on the problem you're solving. Please, let me know.

Reply:
Hi @mariajgrimaldi.
We discussed it internally and decided to go with the second option: we'll update `status` in favor of the `is_passing=True|False`.

Reply:
@mariajgrimaldi we've refactored as discussed. Please, review. 
@GlugovGrGlib fyi.